plugins/modules/storagebox_subaccount.py

```diff
@@ -0,0 +1,473 @@
+        type: str
+        required: false
+
+      samba:
+        description:
+          - Enable or disable Samba.
+        type: bool
+        default: false
+        required: false
+
+      ssh:
+        description:
+          - Enable or disable SSH access.
+        type: bool
+        default: false
+        required: false
+
+      external_reachability:
+        description:
+          - Enable or disable external reachability.
+        type: bool
+        default: false
+        required: false
+
+      webdav:
+        description:
+          - Enable or disable WebDAV.
+        type: bool
+        default: false
+        required: false
+
+      readonly:
+        description:
+          - Enable or disable read-only mode.
+        type: bool
+        default: false
+        required: false
+
+      comment:
+        description:
+          - A custom comment for the subaccount.
+        type: str
+        default: ""
+        required: false
+      idempotence:
+        description:
+          - Select which attribute to use to check subaccount existence.
+          - C(username) or C(comment).
+        type: str
+        choices: [username, comment]
+        default: username
+        required: false
+
+notes:
+  - The module is fully idempotent and supports check mode.
+  - Passwords cannot be compared or diffed, so changes to passwords will always appear as changed in check mode and will not show diffs.
+  - When passwords are autogenerated by the API (by omitting the C(password) field), the resulting password is returned.
+"""
+
+EXAMPLES = r"""
+---
+- name: Create a new subaccount with random password
+  community.hrobot.storagebox_subaccount:
+    storagebox_id: 123456
+    subaccount:
+      homedirectory: "/backups/project1"
+      samba: true
+      ssh: true
+      webdav: false
+      comment: "Backup for Project 1"
+
+- name: Create a subaccount with custom password
+  community.hrobot.storagebox_subaccount:
+    storagebox_id: 123456
+    subaccount:
+      username: "backup1"
+      password: "s3cretPass123"
+      homedirectory: "/data/backup1"
+      readonly: false
+      samba: true
+      ssh: false
+
+- name: Update an existing subaccount
+  community.hrobot.storagebox_subaccount:
+    storagebox_id: 123456
+    subaccount:
+      state: present
+      username: "backup1"
+      homedirectory: "/data/backup1-updated"
+      readonly: true
+      comment: "Updated path and readonly mode"
+
+- name: Delete a subaccount
+  community.hrobot.storagebox_subaccount:
+    storagebox_id: 123456
+    subaccount:
+      state: absent
+      username: "backup1"
+
+- name: Change password for a subaccount
+  community.hrobot.storagebox_subaccount:
+    storagebox_id: 123456
+    subaccount:
+      state: present
+      username: "backup1"
+      password: "n3wSecur3Pass"
+"""
+
+RETURN = r"""
+created:
+  description: Whether a new subaccount was created.
+  type: bool
+  returned: always
```

Comment:
Since this isn't returned when the module fails, it should say "success" and not "always".
```suggestion
  returned: success
```
(Same for the other return values.)